File: examples/sum-of-squares/src/main/java/com/palominolabs/benchpress/example/sumofsquares/SumOfSquaresWorkerConfig.java
package com.palominolabs.benchpress.example.sumofsquares;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Config for an individual worker
 */
class SumOfSquaresWorkerConfig {
    @JsonProperty("start")
    final int first;
    @JsonProperty("end")
    final int last;

    @JsonCreator
    SumOfSquaresWorkerConfig(@JsonProperty("start") int first,
            @JsonProperty("end") int last) {
        this.first = first;
        this.last = last;
    }
}

File: examples/sum-of-squares/src/main/kotlin/com/palominolabs/benchpress/example/sumofsquares/SumOfSquaresWorkerConfig.kt
package com.palominolabs.benchpress.example.sumofsquares;

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Config for an individual worker
 */
class SumOfSquaresWorkerConfig(@param:JsonProperty("start") @field:JsonProperty("start") val first: Long,
                               @param:JsonProperty("end") @field:JsonProperty("end") val last: Long)
